### ISSUE TITLE
renderButtonText renders '[object Object]' if there is a call to setState in onSelect or renderButtonText functions

### ISSUE THREAD

**Issue by Culzean** [sohobloo/react-native-modal-dropdown#139]
I have version 0.6.1 and am using a slightly more complex data structure.

```
         <ModalDropdown
              dropdownStyle={styles.pickerItemStyle}
              style={styles.dropDwon}
              textStyle = {styles.DropDownText}
              dropdownTextStyle = {styles.DropDownPickerText}
              renderButtonText = { (rowData) => {
                this.setState({selectedService: rowData.ssid});
                return rowData.ssid;
                } }
              defaultValue = {"Please select wifi"}
              options = {this.state.services}
              renderRow = {this._renderDropdownRow.bind(this)}
              onSelect = {this.pickerOnSelect.bind(this)}
            >
         </ModalDropdown>
```

If there is a call to setState for both of these functions then the dropdown text renders as [object Object]. If I remove the call then the dropdown renders the ssid strings as expected.

**Comment by Culzean** [comment]
it seems any call to setState is breaking the renderButtonText. [object Object] looks very broken.

**Comment by brascene** [comment]
@Culzean the same here, if I use setState it goes in some endless loop and I receive maximum call stack size exceeded error.

**Comment by mtimofiiv** [comment]
Having the same issue as well, and I am not using any `this.setState` in it either. Here's a snippet:

```jsx
<ModalDropdown
  ref={self => { this.accountPicker = self }}
  defaultValue={i18n.t('ui.inputs.account_picker.placeholder')}
  options={accounts.data}
  onSelect={({ address }) => this.updateTxField({ from: address })}
  renderRow={({ title }) => <Text>{title}</Text>}
  renderButtonText={({ title }) => title}
  style={style.dropdownContainer}
  textStyle={style.dropdownText}
  dropdownStyle={style.dropdownStyle}
/>
```

In my example, `accounts.data` is an array of objects with obviously 2 props, `title` and `address`.

The results are as follows (image 1 is during selection, 2 is after the option is clicked):

<img width="157" alt="screen shot 2018-03-06 at 14 42 12" src="https://user-images.githubusercontent.com/428948/37035367-a1658828-214c-11e8-9eb9-845cf73654b1.png">
<img width="143" alt="screen shot 2018-03-06 at 14 42 21" src="https://user-images.githubusercontent.com/428948/37035368-a185e046-214c-11e8-9ecc-e12feaa1465b.png">

I tried reverting back to version 0.5.0 to see what would happen and in that, `[Object object]` shows up all the time, not just after selection.

**Comment by mtimofiiv** [comment]
I believe the issue is here:

https://github.com/sohobloo/react-native-modal-dropdown/blob/master/components/ModalDropdown.js#L182

It should be using the `renderButtonText` there but it does not seem to. I can take a closer look and submit a PR, need to fix this for my app anyways!

**Comment by mtimofiiv** [comment]
I submitted a PR -> https://github.com/sohobloo/react-native-modal-dropdown/pull/142

**Comment by sohobloo** [comment]
Fixed in v0.6.2 #146

**Comment by Culzean** [comment]
Just testing out the update now. Issue resolved. Thanks!
